server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -57,17 +60,32 @@
 import org.elasticsearch.search.slice.SliceBuilder;
 import org.elasticsearch.search.sort.SortAndFormats;
 import org.elasticsearch.search.suggest.SuggestionSearchContext;
+import org.elasticsearch.threadpool.ThreadPool;
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ThreadPoolExecutor;
 import java.util.function.LongSupplier;
 
 final class DefaultSearchContext extends SearchContext {
 
+    /**
+     * A feature flag to indicate if searches should be run in concurrent mode. Defaults to false.
+     */
+    private static final String CONCURRENT_SEARCH_SYSTEM_PROPERTY = "es.concurrent_search";
+    private static final Boolean CONCURRENT_SEARCH_FEATURE_FLAG;
+    static {
+        final Boolean useConcurrentSearch = Booleans.parseBoolean(System.getProperty(CONCURRENT_SEARCH_SYSTEM_PROPERTY), false);
+        if (useConcurrentSearch && Build.CURRENT.isSnapshot() == false) {
```

Comment:
I think we should remove this condition. This isn't really a "feature flag" in the traditional sense. This condition basically makes it impossible to enable this in release builds, which is what we will want for serverless.